contracts/clients/src/BlsWalletWrapper.ts

```diff
@@ -148,7 +150,11 @@ export default class BlsWalletWrapper {
       blsWalletSigner,
       privateKey,
       await BlsWalletWrapper.BLSWallet(privateKey, verificationGateway),
+      verificationGateway.address,
     );
+    blsWalletWrapper.blockGasLimit = (
+      await provider.getBlock("latest")
+    ).gasLimit;
```

Comment:
With EIP-1559, gas limit can vary between blocks from 15 - 30 million gas. Not sure if this applies to current optimistic rollups, but if it can be variable we should also update this [every block](https://docs.ethers.org/v5/api/providers/provider/#Provider--events).

In constructor:
```typescript
// since this is async, we also need to drop any earlier block queries that are still processing
// and use the latest so older gas limit values don't overwrite new ones.
provider.on("block", async (blockNumber) => {
    const { gasLimit } = await provider.getBlock(blockNumber);
    this.blockGasLimit = gasLimit;
});
// We may need to add a cleanup/stop method to wrapper class to prevent event listener leaks.
```

Reply:
@jacque006 I think that's wrong. It's not the block gas limit that varies in that kind of range but the block gas *used*. The limit is currently fixed at 30m. I wrote a script to look at some random samples to help confirm:

```js
(async () => {
    const latestBlock = await ethereum.request({ method: 'eth_getBlockByNumber', params: ['latest', false] });
    console.log(latestBlock);

    const rangeStart = 15537393; // First post-merge block
    const rangeEnd = 16530455; // Current latest

    for (let i = 0; i < 100; i++) {
        const sampleBlockNo = rangeStart + Math.floor((rangeEnd-rangeStart) * Math.random());
        const block = await ethereum.request({ method: 'eth_getBlockByNumber', params: ['0x'+sampleBlockNo.toString(16), false] });

        console.log(`Sample ${i} | Block ${sampleBlockNo} | Used: ${Number(block.gasUsed)} | Limit: ${Number(block.gasLimit)}`);
    }
})();
```

![Screen Shot 2023-02-01 at 11 46 44 am](https://user-images.githubusercontent.com/9291586/215917496-6d19ff2f-9e52-4e77-8db3-7fd43c04e31d.png)